File: tests/fixtures/resend/resend-webhook-signature-fixed/01-verify-first.ts
import { Resend } from 'resend';
import type { NextRequest } from 'next/server';

const resend = new Resend(process.env.RESEND_API_KEY!);

export async function POST(req: NextRequest) {
  const payload = await req.text();
  resend.webhooks.verify({
    payload,
    headers: {
      id: req.headers.get('svix-id')!,
      timestamp: req.headers.get('svix-timestamp')!,
      signature: req.headers.get('svix-signature')!,
    },
    webhookSecret: process.env.RESEND_WEBHOOK_SECRET!,
  });
  return new Response('ok');
}

File: tests/fixtures/resend/resend-webhook-signature-fixed/01-verify-first.py
import json
import os
import resend
from flask import Flask, jsonify, request

app = Flask(__name__)
resend.api_key = os.environ["RESEND_API_KEY"]


@app.route("/webhook", methods=["POST"])
def handle_webhook():
    payload = request.get_data(as_text=True)
    resend.Webhooks.verify(
        {
            "payload": payload,
            "headers": {
                "id": request.headers.get("svix-id"),
                "timestamp": request.headers.get("svix-timestamp"),
                "signature": request.headers.get("svix-signature"),
            },
            "webhook_secret": os.environ["RESEND_WEBHOOK_SECRET"],
        }
    )
    event = json.loads(payload)
    return jsonify({"received": True, "type": event.get("type")})
